groups/bsl/bslstl/bslstl_optional.h

```diff
@@ -765,787 +1431,10708 @@ void Optional_DataImp<TYPE>::emplace(
+                                   ,
+                                   BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_09)
+                                   );
+#endif  // BSLSTL_OPTIONAL_VARIADIC_LIMIT_D >= 9
+
+#if BSLSTL_OPTIONAL_VARIADIC_LIMIT_D >= 10
+    template <class INIT_LIST_TYPE, class ARGS_01,
+                                    class ARGS_02,
+                                    class ARGS_03,
+                                    class ARGS_04,
+                                    class ARGS_05,
+                                    class ARGS_06,
+                                    class ARGS_07,
+                                    class ARGS_08,
+                                    class ARGS_09,
+                                    class ARGS_10>
+    void emplace(std::initializer_list<INIT_LIST_TYPE>,
+                                   BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_01)
+                                   ,
+                                   BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_02)
+                                   ,
+                                   BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_03)
+                                   ,
+                                   BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_04)
+                                   ,
+                                   BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_05)
+                                   ,
+                                   BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_06)
+                                   ,
+                                   BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_07)
+                                   ,
+                                   BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_08)
+                                   ,
+                                   BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_09)
+                                   ,
+                                   BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_10)
+                                   );
+#endif  // BSLSTL_OPTIONAL_VARIADIC_LIMIT_D >= 10
+
+#endif
+#else
+// The generated code below is a workaround for the absence of perfect
+// forwarding in some compilers.
+    template <class... ARGS>
+    void emplace(BSLS_COMPILERFEATURES_FORWARD_REF(ARGS)...);
+
+#if defined(BSLS_COMPILERFEATURES_SUPPORT_GENERALIZED_INITIALIZERS)
+    template <class INIT_LIST_TYPE, class... ARGS>
+    void emplace(std::initializer_list<INIT_LIST_TYPE>,
+                                        BSLS_COMPILERFEATURES_FORWARD_REF(ARGS)
+                                        ...);
+#endif
+// }}} END GENERATED CODE
+#endif
+    void reset() BSLS_KEYWORD_NOEXCEPT;
+        // Reset this object to the default constructed state (i.e., to a
+        // disengaged state).
+
+    void swap(optional& other);
+        // Efficiently exchange the value of this object with the value of the
+        // specified 'other' object.  In effect, performs
+        // 'using bsl::swap; swap(c, other.c);'.
```

Comment:
We might have UB here for allocator-aware types which is worth documenting.  Please take a look at `NullableValue` doc for member swap

Reply:
- copied the documentation from Nullablevalue to AA optional's swap and adapted it as necessary. 
 - copied the documentation from Nullablevalue to NonAA optional's swap for consistency, and removed the allocator note
 - copied the documentation from Nullablevalue to free function swaps (both bsl::optional and std::optional overloads) and adapted it as necessary.